source/2_platform/rex_windows/src/console_application.cpp

```diff
@@ -0,0 +1,114 @@
+#include "rex_windows/console_application.h"
+
+#include "rex_engine/diagnostics/logging/log_macros.h"
+#include "rex_engine/event_system.h"
+
+#include <Windows.h>
+
+namespace rex
+{
+  namespace win32
+  {
+    DEFINE_LOG_CATEGORY(LogWinConsoleApp, rex::LogVerbosity::Log);
+
+    CoreApplication* g_this_app = nullptr; // NOLINT(cppcoreguidelines-avoid-non-const-global-variables)
```

Comment:
What is the difference between this ptr and the app_instance ptr?
Can't we just use one ptr?

Reply:
there is no app instance anymore.

Reply:
Why not?

Reply:
because it wasn't needed at the time, so I removed it.
if we want a generic design back, that's fine, but that's out of scope of this change imo

Reply:
So to make things clear we had a feature that worked in the long run and you removed that and replaced it with something that works short term, isn't that against everything you are telling me?

Reply:
we discussed together if we should remove it or not.
I believe I asked you if you were okay with removing it as we didn't see a point in having it (it's too exposed as it can be accessed everywhere) so we deleted it